lib/web/servers.go

```diff
@@ -81,6 +81,25 @@ func (h *Handler) clusterKubePodsGet(w http.ResponseWriter, r *http.Request, p h
 	}, nil
 }
 
+// clusterKubeResourcesGet returns supported requested kubernetes subresources eg: pods, namespaces, secrets etc.
+func (h *Handler) clusterKubeResourcesGet(w http.ResponseWriter, r *http.Request, p httprouter.Params, sctx *SessionContext, site reversetunnelclient.RemoteSite) (interface{}, error) {
+	clt, err := sctx.NewKubernetesServiceClient(r.Context(), h.cfg.ProxyWebAddr.Addr)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	resp, err := listKubeResources(r.Context(), clt, r.URL.Query(), site.GetName(), r.URL.Query().Get("kind"))
```

Comment:
can we add a validation on kind?

if kind==""{
return trace.BadParameter..
}

 if !slices.Contains(types.KubernetesResources, kind){
return trace.BadParameter(kind is not valid, valid kinds %v, types.KubernetesResources)
}

Reply:
done, i also added a check for empty kube cluster since that's also required